atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/Schema.java

```diff
@@ -109,6 +112,19 @@ public void addTableDefinition(String tableName, TableDefinition definition) {
         Preconditions.checkArgument(
                 Schemas.isTableNameValid(tableName),
                 "Invalid table name " + tableName);
+        if (!ignoreTableNameLengthChecks) {
+            int internalTableNameLength = Schemas.getInternalTableNameLength(tableName, namespace);
+            Preconditions.checkArgument(
+                    internalTableNameLength < POSTGRES_TABLE_NAME_CHAR_LIMIT,
+                    "Table name %s is too long, exceeds Cassandra and Postgres limits. " +
+                            "If running using a different KVS, set the ignoreTableNameLength flag.",
```

Comment:
The Cassandra check should be before the postgres one as that limit is lower, to ensure that users are warned with the appropriate message.